pam/gdm/conversation.go

```diff
@@ -26,3 +27,137 @@ func sendToGdm(pamMTx pam.ModuleTransaction, data []byte) ([]byte, error) {
 	defer binRes.Release()
 	return binRes.Decode(decodeJSONProtoMessage)
 }
+
+// Send sends the data to the PAM Module, returning the JSON data.
+func (d *Data) Send(pamMTx pam.ModuleTransaction) ([]byte, error) {
+	bytes, err := d.JSON()
+	if err != nil {
+		return nil, err
+	}
+
+	return sendToGdm(pamMTx, bytes)
+}
+
+// SendParsed sends the data to the PAM Module and returns the parsed Data.
+func (d *Data) SendParsed(pamMTx pam.ModuleTransaction) (*Data, error) {
+	bytes, err := d.JSON()
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Println("Sending to GDM", string(bytes))
+	jsonValue, err := sendToGdm(pamMTx, bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	gdmData, err := NewDataFromJSON(jsonValue)
+	fmt.Println("Got from GDM", string(jsonValue))
+	if err != nil {
+		return nil, err
+	}
+	return gdmData, nil
+}
```

Comment:
We are parsing the response, not the data that is being sent, so having this as a separate function feels confusing and unnecessary.

I think we should instead only have the `Send` function and then use the NewDataFromJSON function when we need to parse the response from GDM.

Reply:
Well, that would really make the code very repetitive everywhere. In fact the data we expect to receive is always a JSON in `Data` format. Doing this manually everywhere would duplicate lots of code in the model making things really hard to read and maintain IMHO. And also very prone to errors.

Reply:
I've renamed it to just `SendData` while the other without checks is private. I can probably get rid of it but I liked to do incremental testing.